src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -969,6 +969,105 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/index:
+    post:
+      tags:
+        - snapshots
+        - repository
+      description: Create a new search index for a snapshot
+      operationId: createSearchIndex
+      parameters:
+        - name: id
+          in: path
+          description: A UUID to used to identify an object in the repository
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Index to create from SQL
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SearchIndexRequest'
+        required: true
+      responses:
+        200:
+          description: Return index result
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SearchIndexModel'
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see snapshot
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        404:
+          description: Not found - snapshot id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/query:
+    post:
+      tags:
+        - snapshots
+        - repository
+      description: Search against one or more snapshots
```

Comment:
The endpoint is `query` so I might use `query` in these descriptions instead of `search`. E.g. "Run a query on the metadata of one or more snapshots"